.travis.yml

```diff
@@ -7,6 +7,8 @@ cache:
   directories:
     - $HOME/.cabal/packages
     - $HOME/.cabal/store
+    - $HOME/llvm-3.9
+    - $HOME/llvm-3.9.0rc2.src
```

Comment:
I don't think caching the source is doing anything. Travis doesn't have any magic to skip commands, it just saves what's in the cache between runs. By cache source but downloading it unconditionally, you download it an extra time.

I recommend you cache only the built llvm, and also cache it one level deeper so these lines aren't dependent on the llvm version.